### PR TITLE
Apollo Refactor Phase 2: Temporal workflow integration

### PR DESCRIPTION
**Pull request by rockythorn**
# Updates
.gitignore:
- Exclude additional development and log directories (venv, container_data, personal_work_dir, *.log) to keep the repository clean.

apollo/rhcsaf/__init__.py:
- Added extract_rhel_affected_products_for_db to extract affected products from the CSAF product tree, expanding 'noarch' and mapping architecture names.
- Refactored red_hat_advisory_scraper to accept a CSAF dictionary directly instead of a file path, and to use the new extraction function.
- Updated logic for extracting Bugzilla IDs
- Improved logging for missing product trees and architectures.

apollo/rhworker/poll_rh_activities.py:
- Refactored process_csaf_file to accept a CSAF dictionary and improved logging.
- Switched CSAF processing to stream JSON files directly from Red Hat using CSV indexes, filtering advisories to those updated within the last 30 days.
- Added checks and logs for advisories with no fixed packages or affected products.
- Refactored process_csaf_files for streaming and parallelization groundwork.

apollo/rpmworker/BUILD.bazel:
- Added dependency on //apollo/rpm_helpers:rpm_helpers_lib to ensure shared RPM parsing logic is available.

apollo/rpmworker/repomd.py:
- Refactored to use parse_nevra helper for NEVRA string parsing, reducing code duplication.

apollo/rpmworker/rh_matcher_activities.py:
- Refactored to use parse_nevra for NEVRA parsing.
- Improved logging throughout advisory cloning and matching.
- Always constructs and sets the topic for new advisories.
- Clarified debug messages for package matching and advisory processing.

apollo/server/BUILD.bazel:
- Added dependency on //apollo/rpm_helpers:rpm_helpers_lib for shared RPM parsing logic.

apollo/tests/test_csaf_processing.py:
- Switched to IsolatedAsyncioTestCase for async test support.
- Added and improved tests for process_csaf_file and process_csaf_files, including error and edge cases, using mocks for async DB and network calls.

apollo/tests/test_rhcsaf.py:
- Added tests for extract_rhel_affected_products_for_db, covering basic extraction, missing product_tree, unknown architecture, noarch expansion, missing CPE, and major-only version parsing.
- Improved coverage for red_hat_advisory_scraper, ensuring None is returned for invalid input.

These changes improve the reliability, maintainability, and observability of the advisory processing pipeline, and lay the groundwork for future parallelization of CSAF document processing.

# Testing
This PR includes comprehensive validation of advisory data between the Pull Request (development) and database using the RESF `master` branch code. The following types of checks and comparisons are performed:

Validation and Comparison Performed
## Advisory Presence:

Ensures that all advisories expected in both environments are present, and identifies any advisories that exist only in one environment.

#### Advisories generated only from Pull Request code (81):
(This indicates these are not available from the HYDRA API)
- `RLSA-2025:4048`
- `RLSA-2025:7418`
- `RLSA-2025:3407`
- `RLSA-2025:4487`
- `RLSA-2025:7428`
- `RLSA-2025:3408`
- `RLSA-2025:3683`
- `RLBA-2025:2991`
- `RLSA-2025:4658`
- `RLSA-2025:3082`
- `RLSA-2025:3113`
- `RLSA-2025:3210`
- `RLSA-2025:3974`
- `RLSA-2025:3828`
- `RLSA-2025:7416`
- `RLSA-2025:3645`
- `RLSA-2025:3582`
- `RLSA-2025:3260`
- `RLSA-2025:3713`
- `RLSA-2025:7589`
- `RLSA-2025:7686`
- `RLSA-2025:3631`
- `RLSA-2025:7540`
- `RLSA-2025:3262`
- `RLSA-2025:3845`
- `RLSA-2025:3833`
- `RLSA-2025:3264`
- `RLSA-2025:3344`
- `RLSA-2025:4797`
- `RLSA-2025:4244`
- `RLSA-2025:7539`
- `RLSA-2025:4461`
- `RLSA-2025:3027`
- `RLSA-2025:4787`
- `RLSA-2025:4560`
- `RLSA-2025:3026`
- `RLSA-2025:4051`
- `RLSA-2025:4459`
- `RLSA-2025:7895`
- `RLSA-2025:7894`
- `RLSA-2025:3388`
- `RLSA-2025:4492`
- `RLSA-2025:4043`
- `RLSA-2025:3421`
- `RLSA-2025:3855`
- `RLSA-2025:3615`
- `RLSA-2025:3997`
- `RLSA-2025:4063`
- `RLSA-2025:4443`
- `RLSA-2025:3772`
- `RLSA-2025:3411`
- `RLSA-2025:3634`
- `RLSA-2025:4049`
- `RLSA-2025:7432`
- `RLSA-2025:4488`
- `RLSA-2025:3406`
- `RLSA-2025:3617`
- `RLSA-2025:7571`
- `RLSA-2025:3261`
- `RLSA-2025:4491`
- `RLSA-2025:4597`
- `RLSA-2025:4341`
- `RLSA-2025:4791`
- `RLSA-2025:7426`
- `RLSA-2025:4170`
- `RLSA-2025:3531`
- `RLSA-2025:7433`
- `RLSA-2025:3773`
- `RLSA-2025:4362`
- `RLSA-2025:7387`
- `RLSA-2025:7417`
- `RLSA-2025:4460`
- `RLSA-2025:3913`
- `RLSA-2025:4458`
- `RLSA-2025:4493`
- `RLSA-2025:7569`
- `RLSA-2025:3893`
- `RLSA-2025:3852`
- `RLSA-2025:4263`
- `RLSA-2025:3894`
- `RLSA-2025:7435`

#### Advisories generated only from RESF `master` code (70)
(These are not available in the CSAF files produced by Red Hat and don't have associated CVEs)

- `RLBA-2025:1336`
- `RLBA-2025:2609`
- `RLBA-2025:1817`
- `RLBA-2025:0740`
- `RLBA-2025:2611`
- `RLBA-2025:0924`
- `RLBA-2025:0909`
- `RLBA-2025:2610`
- `RLBA-2025:2607`
- `RLBA-2025:0728`
- `RLBA-2025:0418`
- `RLBA-2025:0730`
- `RLBA-2025:2353`
- `RLBA-2025:2592`
- `RLBA-2025:0731`
- `RLBA-2025:1316`
- `RLBA-2025:2595`
- `RLBA-2025:2596`
- `RLBA-2025:0935`
- `RLEA-2025:0734`
- `RLBA-2025:0515`
- `RLBA-2025:1682`
- `RLBA-2025:0927`
- `RLBA-2025:2605`
- `RLBA-2025:2598`
- `RLBA-2025:0779`
- `RLBA-2025:0735`
- `RLBA-2025:2603`
- `RLBA-2025:0748`
- `RLBA-2025:0541`
- `RLBA-2025:0913`
- `RLBA-2025:0919`
- `RLBA-2025:1104`
- `RLEA-2025:1337`
- `RLBA-2025:0732`
- `RLBA-2025:2604`
- `RLBA-2025:2601`
- `RLBA-2025:0727`
- `RLBA-2025:0921`
- `RLBA-2025:2617`
- `RLBA-2025:1573`
- `RLBA-2025:2608`
- `RLBA-2025:0573`
- `RLBA-2025:0930`
- `RLBA-2025:2352`
- `RLBA-2025:2597`
- `RLBA-2025:0926`
- `RLBA-2025:1240`
- `RLBA-2025:2594`
- `RLBA-2025:2602`
- `RLBA-2025:0916`
- `RLBA-2025:0729`
- `RLBA-2025:2590`
- `RLBA-2025:0928`
- `RLBA-2025:1344`
- `RLBA-2025:2618`
- `RLBA-2025:0911`
- `RLBA-2025:0572`
- `RLBA-2025:2871`
- `RLBA-2025:0744`
- `RLBA-2025:2599`
- `RLBA-2025:2593`
- `RLEA-2025:0738`
- `RLBA-2025:2606`
- `RLBA-2025:0742`
- `RLBA-2025:2591`
- `RLBA-2025:0745`
- `RLBA-2025:0929`
- `RLBA-2025:0741`
- `RLBA-2025:2613`

## Field-by-Field Comparison:

For advisories present in both databases, each field (excluding IDs and timestamps) is compared to detect any differences in values.

The differences shown in this file come from a different set or order of packages listed in the fields for each advisory.

The rest of the topic text is identical, indicating that the main content difference is the package list. This suggests the advisories are functionally similar, but the presentation or grouping of affected packages differs between the Pull Request and RESF Master environments.

`RLSA-2025:0595`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for module.redis, redis.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for redis, module.redis.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:0733`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| synopsis | Moderate: bzip2 security and bug fix update | Moderate: bzip2 security update |
| description | The bzip2 packages contain a freely available, high-quality data compressor. It provides both standalone compression and decompression utilities, as well as a shared library for use with other programs. Security Fix(es):* bzip2: bzip2: Data integrity error when decompressing (with data integrity tests fail). (CVE-2019-12900)* bzip2: bzip2: Data integrity error when decompressing (with data integrity tests fail). [rhel-8.10.z] (CVE-2019-12900)Bug Fix(es):* Recent bzip2 Rocky Linux8 update breaks data integrity tests (JIRA:Rocky Linux-67824)For more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer the CVE page(s) listed in the References section. | The bzip2 packages contain a freely available, high-quality data compressor. It provides both standalone compression and decompression utilities, as well as a shared library for use with other programs. Security Fix(es):* bzip2: bzip2: Data integrity error when decompressing (with data integrity tests fail). (CVE-2019-12900)For more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section. |

`RLSA-2025:0737`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for mariadb, module.mariadb, galera, module.galera, module.Judy, Judy.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for module.mariadb, Judy, galera, module.galera, module.Judy, mariadb.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:0739`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for mariadb, module.mariadb, galera, module.galera, module.Judy, Judy.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for module.mariadb, Judy, galera, module.galera, module.Judy, mariadb.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:0746`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for gimp, module.python2-pycairo, python2-pycairo, module.gimp, module.pygobject2, pygobject2, module.pygtk2, pygtk2.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for gimp, module.gimp, module.python2-pycairo, pygobject2, python2-pycairo, pygtk2, module.pygtk2, module.pygobject2.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:0912`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for module.mariadb, module.galera, galera, mariadb.This update affects Rocky Linux 9.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for module.mariadb, mariadb, galera, module.galera.This update affects Rocky Linux 9.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1351`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for nodejs-packaging, module.nodejs, module.nodejs-packaging, nodejs, nodejs-nodemon, module.nodejs-nodemon.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for nodejs-packaging, nodejs-nodemon, nodejs, module.nodejs, module.nodejs-nodemon, module.nodejs-packaging.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1372`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for containernetworking-plugins, libslirp, aardvark-dns, module.podman, module.fuse-overlayfs, module.runc, podman, container-selinux, runc, module.aardvark-dns, containers-common, module.containernetworking-plugins, module.python-podman, module.crun, cockpit-podman, udica, module.buildah, fuse-overlayfs, module.cockpit-podman, module.toolbox, module.container-selinux, criu, toolbox, module.libslirp, module.netavark, module.udica, module.containers-common, module.skopeo, module.criu, buildah, crun, python-podman, slirp4netns, module.oci-seccomp-bpf-hook, module.slirp4netns, skopeo, module.conmon, conmon, oci-seccomp-bpf-hook, netavark.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for module.libslirp, udica, module.criu, module.runc, oci-seccomp-bpf-hook, module.podman, podman, module.skopeo, module.python-podman, containers-common, netavark, containernetworking-plugins, libslirp, crun, module.udica, module.slirp4netns, slirp4netns, module.fuse-overlayfs, module.conmon, module.aardvark-dns, skopeo, module.toolbox, module.oci-seccomp-bpf-hook, aardvark-dns, buildah, fuse-overlayfs, toolbox, module.containers-common, python-podman, module.containernetworking-plugins, module.netavark, container-selinux, runc, cockpit-podman, conmon, module.crun, module.cockpit-podman, criu, module.container-selinux, module.buildah.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1443`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for nodejs-packaging, module.nodejs-packaging, nodejs-nodemon, module.nodejs-nodemon.This update affects Rocky Linux 9.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for nodejs-nodemon, nodejs-packaging, module.nodejs-nodemon, module.nodejs-packaging.This update affects Rocky Linux 9.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1446`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for nodejs-packaging, module.nodejs-packaging, nodejs-nodemon, module.nodejs-nodemon.This update affects Rocky Linux 9.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for nodejs-nodemon, nodejs-packaging, module.nodejs-nodemon, module.nodejs-packaging.This update affects Rocky Linux 9.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1582`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for nodejs-packaging, module.nodejs-packaging, module.nodejs, nodejs, nodejs-nodemon, module.nodejs-nodemon.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for nodejs-packaging, nodejs-nodemon, nodejs, module.nodejs, module.nodejs-nodemon, module.nodejs-packaging.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1611`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for nodejs-packaging, module.nodejs, module.nodejs-packaging, nodejs, nodejs-nodemon, module.nodejs-nodemon.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for nodejs-packaging, nodejs-nodemon, nodejs, module.nodejs, module.nodejs-nodemon, module.nodejs-packaging.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1613`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for nodejs-packaging, module.nodejs-packaging, nodejs-nodemon, module.nodejs-nodemon.This update affects Rocky Linux 9.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for nodejs-nodemon, nodejs-packaging, module.nodejs-nodemon, module.nodejs-packaging.This update affects Rocky Linux 9.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1673`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for mysql, module.mecab, mecab, module.mecab-ipadic, mecab-ipadic, module.mysql.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for mecab, module.mecab, mysql, module.mecab-ipadic, module.mysql, mecab-ipadic.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1736`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for module.postgresql, postgresql, module.postgres-decoderbufs, pgaudit, pg_repack, postgres-decoderbufs, module.pgaudit, module.pg_repack.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for module.pgaudit, postgresql, postgres-decoderbufs, module.postgresql, pg_repack, module.pg_repack, pgaudit, module.postgres-decoderbufs.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1739`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for module.postgresql, postgresql, module.postgres-decoderbufs, pgaudit, pg_repack, postgres-decoderbufs, module.pgaudit, module.pg_repack.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for postgresql, module.pgaudit, postgres-decoderbufs, module.postgresql, pg_repack, module.pg_repack, pgaudit, module.postgres-decoderbufs.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1740`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for module.postgresql, postgresql, module.postgres-decoderbufs, pgaudit, pg_repack, postgres-decoderbufs, module.pgaudit, module.pg_repack.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for module.pgaudit, postgresql, postgres-decoderbufs, module.postgresql, pg_repack, module.pg_repack, pgaudit, module.postgres-decoderbufs.This update affects Rocky Linux 8.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1741`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for module.postgresql, postgresql, module.postgres-decoderbufs, pgaudit, pg_repack, postgres-decoderbufs, module.pgaudit, module.pg_repack.This update affects Rocky Linux 9.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for module.pgaudit, postgresql, postgres-decoderbufs, module.postgresql, pg_repack, module.pg_repack, pgaudit, module.postgres-decoderbufs.This update affects Rocky Linux 9.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

`RLSA-2025:1743`

| Field | Pull Request Value | RESF Master Value |
|-------|--------------------|-------------------|
| topic | An update is available for module.postgresql, postgresql, module.postgres-decoderbufs, pgvector, pgaudit, pg_repack, postgres-decoderbufs, module.pgaudit, module.pg_repack, module.pgvector.This update affects Rocky Linux 9.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list | An update is available for pgvector, postgresql, module.pgaudit, postgres-decoderbufs, module.postgresql, module.pgvector, pg_repack, module.pg_repack, pgaudit, module.postgres-decoderbufs.This update affects Rocky Linux 9.A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability from the CVE list |

## Related Data Consistency:

For each shared advisory, the number of associated packages, CVEs, and bugs is counted and compared between environments.
Any discrepancies in these counts are highlighted, ensuring that related data is consistent and complete.
Note: There were 99 advisories shared between both advisory generation runs and of those 12 had differences in either the number of associated packages, number of associated CVEs, or number of associated Bugzilla fixes.

| Advisory Name | Pull Request Packages | RESF Master Packages | Pull Request CVEs | RESF Master CVEs | Pull Request Bugs | RESF Master Bugs |
|--------------|-------------|--------------|----------|-----------|----------|-----------|
| `RLSA-2025:0210` | 26 | 24 | 1 | 1 | 1 | 1 |
| `RLSA-2025:0281` | 8 | 8 | 7 | 7 | 7 | 2 |
| `RLSA-2025:0288` | 63 | 63 | 1 | 1 | 1 | 0 |
| `RLSA-2025:0422` | 68 | 68 | 1 | 1 | 1 | 0 |
| `RLSA-2025:0426` | 68 | 68 | 1 | 1 | 1 | 0 |
| `RLSA-2025:0936` | 8 | 5 | 1 | 1 | 1 | 1 |
| `RLSA-2025:2867` | 41 | 40 | 1 | 1 | 1 | 1 |

The differences in number of packages appears to be due to the fact that the new code is adding `src.rpm` packages for more architectures than the original code.

The differences in number of associated bugs is due to the new code properly pulling in associated bugzilla ticket IDs for each CVE where the existing code is not pulling this as reliably from the HYDRA API.

### CODE REVIEW

File: apollo/tests/test_csaf_processing.py
Comment (rockythorn) [inline apollo/tests/test_csaf_processing.py:1]: Key changes:

* Imports and uses `initialize_test_db` and `close_test_db` for async class-level setup/teardown.
* Removes all mocking of database models—tests now use real Tortoise ORM models and database operations.
* Cleans up all tables in `tearDown` to ensure test isolation.
* Test methods now verify actual database state, not just mock calls.
* Only the logger is mocked; all DB interactions are real.
* This approach makes tests simpler, more maintainable, and closer to real runtime behavior, while ensuring each test runs with a clean database state.

File: apollo/rhcsaf/__init__.py [hunk 1]
```diff
@@ -8,14 +8,94 @@
 
 logger = Logger()
 
-def red_hat_advisory_scraper(filename: pathlib.Path):
-    logger.info(f"Parsing CSAF document{filename}")
-    with open(filename, "r") as f:
-        csaf = json.load(f)
-    
+def extract_rhel_affected_products_for_db(csaf):
```
Comment (rockythorn) [inline apollo/rhcsaf/__init__.py:11]: This new function is to properly pull needed information from the CSAF documents to fill in the `red_hat_advisory_affected_products` table. The data for this is pulled from the `product_tree` inside the json document.

File: apollo/rhworker/poll_rh_activities.py [hunk 2]
```diff
@@ -406,47 +413,79 @@ async def process_csaf_file(filepath: str) -> Optional[RedHatAdvisory]:
...
+        logger.info("Fetching CSV files from Red Hat")
+        changes = await fetch_csv_with_dates(session, base_url + "changes.csv")
+        releases = await fetch_csv_with_dates(session, base_url + "releases.csv")
+        deletions = await fetch_csv_with_dates(session, base_url + "deletions.csv")
+
+        # Merge changes and releases, keeping the most recent timestamp for each advisory
+        all_advisories = {**changes, **releases}
+        # Remove deletions
+        for advisory_id in deletions:
+            all_advisories.pop(advisory_id, None)
```
Comment (rockythorn) [inline apollo/rhworker/poll_rh_activities.py:460]: We may want to handle this differently. As of now we ignore any advisories that were removed by Red Hat but we may want to actually delete applicable advisories from Apollo. This will need further discussion.

File: apollo/rpmworker/rh_matcher_activities.py [hunk 3]
```diff
@@ -392,21 +395,20 @@ async def clone_advisory(
...
-        if not new_advisory.topic:
-            package_names = list({p.package_name for p in new_pkgs})
-            affected_products = list(
-                {
-                    f"{product.name} {mirror.match_major_version}"
-                    for mirror in mirrors
-                }
-            )
-            topic = f"""An update is available for {', '.join(package_names)}.
+        # Construct topic
```
Comment (rockythorn) [inline apollo/rpmworker/rh_matcher_activities.py:398]: From what I saw the HYDRA API never returns a topic meaning we always construct the topic. CSAF documents DO contain topics but they don't match the formatting we've been using for Rocky advisories. I opted to change this to always construct the topic to keep consistency with what we are already publishing.

**Review by dvdgomez — CHANGES_REQUESTED** [review]

File: apollo/rhcsaf/__init__.py [hunk 4]
```diff
@@ -8,14 +8,94 @@
...
+        for branch in family_branch.get("branches", []):
+            if branch.get("category") == "product_name":
+                prod = branch.get("product", {})
+                prod_name = prod.get("name")
+                cpe = prod.get("product_identification_helper", {}).get("cpe")
+                break
+        if not prod_name or not cpe:
+            continue
+
+        # Parses the CPE string to extract major and minor version numbers
```
[the remark this answers is not in the document]
Comment (rockythorn) [reply apollo/rhcsaf/__init__.py:54]: I'll add an example that shows the original string and how the code parses it.

File: apollo/rhcsaf/__init__.py [hunk 5]
```diff
@@ -8,14 +8,94 @@
...
+    # List of main architectures to expand 'noarch'
+    main_arches = list(arch_name_map.keys())
+    affected_products = set()
+    product_tree = csaf.get("product_tree", {})
+    if not product_tree:
+        logger.warning("No product tree found in CSAF document")
+        return affected_products
+
+    # Iterate over all vendor branches in the product tree
+    for vendor_branch in product_tree.get("branches", []):
```
[the remark this answers is not in the document]
Comment (rockythorn) [reply apollo/rhcsaf/__init__.py:33]: I'm not overly concerned with speed in this case so I'm not too concerned with the multiple iterations.

**Review by jason-rodri — APPROVED** [review]

File: apollo/rhcsaf/__init__.py [hunk 6]
```diff
@@ -8,14 +8,94 @@
...
+        # Find the product_family branch for RHEL
+        family_branch = None
+        for b in vendor_branch.get("branches", []):
+            if b.get("category") == "product_family" and b.get("name") == "Red Hat Enterprise Linux":
+                family_branch = b
```
[the remark this answers is not in the document]
Comment (rockythorn) [reply apollo/rhcsaf/__init__.py:39]: I'll get this updated in the next commit

File: apollo/rhworker/poll_rh_activities.py [hunk 7]
```diff
@@ -264,15 +267,18 @@ async def get_rh_advisories(from_timestamp: str = None) -> None:
...
-    logger.info("INFO logging is enabled")
-    logger.debug("DEBUG logging is enabled")
-    data = red_hat_advisory_scraper(filepath)
+    data = red_hat_advisory_scraper(json_data)
     if not data:
         logger.warning(f"No data returned from scraper for {filepath}")
         return None
+    # Check if advisory has any fixed packages. If not, skip it. It could have no fixed packages
+    # because there were not packages for a Red Hat product starting with "Red Hat Enterprise Linux"
+    if len(data.get("red_hat_fixed_packages")) < 1:
```
[the remark this answers is not in the document]
Comment (rockythorn) [reply apollo/rhworker/poll_rh_activities.py:279]: This make a lot of sense. Simplifies the code.

**Review by dvdgomez — APPROVED** [review]
LGTM!
